pkg/security/security_profile/activity_tree/process_node.go

```diff
@@ -319,9 +333,103 @@ func (pn *ProcessNode) InsertBindEvent(evt *model.Event, generationType NodeGene
 	}
 
 	// Insert bind event
-	if sock.InsertBindEvent(&evt.Bind, generationType, evt.Rules, dryRun) {
+	if sock.InsertBindEvent(&evt.Bind, imageTag, generationType, evt.Rules, dryRun) {
 		newNode = true
 	}
 
 	return newNode
 }
+
+func (pn *ProcessNode) applyImageTagOnLineageIfNeeded(imageTag string) {
+	if imageTag != "" {
+		if !slices.Contains(pn.ImageTags, imageTag) {
+			pn.ImageTags = append(pn.ImageTags, imageTag)
+			parent := pn.GetParent()
+			for parent != nil {
+				parent.AppendImageTag(imageTag)
+				parent = parent.GetParent()
+			}
+		}
+	}
+}
+
+// TagAllNodes tags this process, its files/dns/socks and childrens with the given image tag
+func (pn *ProcessNode) TagAllNodes(imageTag string) {
+	if imageTag == "" {
+		return
+	}
+
+	if !slices.Contains(pn.ImageTags, imageTag) {
+		pn.ImageTags = append(pn.ImageTags, imageTag)
+	}
+	for _, file := range pn.Files {
+		file.tagAllNodes(imageTag)
+	}
+	for _, dns := range pn.DNSNames {
+		dns.appendImageTag(imageTag)
+	}
+	for _, sock := range pn.Sockets {
+		sock.appendImageTag(imageTag)
+	}
+	for _, child := range pn.Children {
+		child.TagAllNodes(imageTag)
+	}
+}
+
+func removeImageTagFromList(imageTags []string, imageTag string) []string {
+	if index := slices.Index(imageTags, imageTag); index != -1 {
+		return slices.Delete(imageTags, index, index+1)
+	}
+	return imageTags
+}
+
+// EvictImageTag will remmove every trace of this image tag, and returns true if the process node should be removed
+// also, recompute the list of dnsnames and syscalls
+func (pn *ProcessNode) EvictImageTag(imageTag string, DNSNames *utils.StringKeys, SyscallsMask map[int]int) bool {
+	if imageTag == "" {
+		return false
+	}
+
+	if !slices.Contains(pn.ImageTags, imageTag) {
+		return false // this node don't have the tag, and all his childs/files/dns/etc shouldn't have neither
+	}
+	pn.ImageTags = removeImageTagFromList(pn.ImageTags, imageTag)
```

Comment:
What about a removeImageTagFromList with the imageTag == "" &&  that will return if it succeeded or not

Reply:
Good idea! I'll do it